app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperManager.kt

```diff
@@ -85,9 +88,35 @@ class WallpaperManager(private val settings: Settings) {
     private fun getCurrentWallpaperFromSettings(): Wallpaper {
         val currentWallpaper = settings.currentWallpaper
         return if (currentWallpaper.isEmpty()) {
-            Wallpaper.NONE
+            defaultWallpaper
         } else {
-            Wallpaper.valueOf(currentWallpaper)
+            wallpapers.firstOrNull { it.name == currentWallpaper } ?: defaultWallpaper
         }
     }
+
+    /**
+     * Loads wallpapers from the storage and populate [wallpapers] with them.
+     */
+    fun loadWallpapers(context: Context) {
+        val assetsManager = context.assets
+        val files = assetsManager.list(WALLPAPERS_DIRECTORY)
+        val loadedWallpapers = files?.map { fileName ->
+            assetsManager.open("$WALLPAPERS_DIRECTORY/$fileName").use {
+                val image = BitmapFactory.decodeStream(it)
+                val isDark = wallpaperToIsDark[fileName] ?: false
+                Wallpaper(
+                    name = fileName,
+                    resource = BitmapDrawable(context.resources, image),
+                    isDark = isDark
+                )
+            }
+        } ?: return
+        wallpapers = (listOf(defaultWallpaper) + loadedWallpapers)
+    }
+
+    companion object {
+        private const val WALLPAPERS_DIRECTORY = "wallpapers"
+        private val wallpaperToIsDark = mapOf("wallpaper_1.png" to true, "wallpaper_2.png" to false)
+        val defaultWallpaper = Wallpaper(name = "homeBackground")
+    }
```

Comment:
It feels just a bit off to me to have an instantiated class as a static property. Primitive data and constants feel like a better fit. Maybe moving the `DEFAULT_RESOURCE` to the companion here and moving the `defaultWallpaper` inside the `WallpaperManager` would make sense.

Another thing you could consider is creating a list of wallpapers within the wallpaper manager. This might be easier to swap out later as well, once we better define how we're going to load the list from disk. For example:

```
class WallpaperManager {
    private val wallpapers = listOf(Wallpaper1, Wallpaper2, Wallpaper3)
}
```
could be later be easily updated to 
```
class WallpaperManager {
    private val wallpapers = loadWallpaperListFromDisk()
}
```

Could even go so far as to define it as an interface now, making it even easier to refactor later and test now:

```
interface WallpaperLoader {
    fun loadWallpaperList(): List<Wallpaper>
}

class StaticListWallpaperLoader() {
    override fun loadWallpaperList(): List<Wallpaper>
}
```

If you went in a direction like that, it could also make sense to include a `fun loadWallpaperImage` in it to retrieve the images from disk as well.